Source/WebCore/style/values/StyleValueTypes.h

```diff
@@ -42,35 +42,31 @@ namespace Style {
 
 class BuilderState;
 
+// Helper for declaring types in the Style namespace as Tuple-Like.
+#define STYLE_TUPLE_LIKE_CONFORMANCE(t, numberOfArguments) \
+    namespace std { \
+        template<> class tuple_size<WebCore::Style::t> : public std::integral_constant<size_t, numberOfArguments> { }; \
+        template<size_t I> class tuple_element<I, WebCore::Style::t> { \
+        public: \
+            using type = decltype(WebCore::Style::get<I>(std::declval<WebCore::Style::t>())); \
+        }; \
+    } \
+    template<> inline constexpr bool WebCore::TreatAsTupleLike<WebCore::Style::t> = true; \
+\
+
 // Types can specialize this and set the value to true to be treated as "non-converting"
 // for css to style / style to css conversion algorithms. This means the type is identical
 // for both CSS and Style systems (e.g. a constant value or an enum).
 template<class> inline constexpr bool TreatAsNonConverting = false;
 
-// Types can specialize this and set the value to true to be treated as "tuple-like"
-// for css to style / style to css conversion algorithms.
-// NOTE: This gets automatically specialized when using the STYLE_TUPLE_LIKE_CONFORMANCE macro.
-template<class> inline constexpr bool TreatAsTupleLike = false;
-
-// Types can specialize this and set the value to true to be treated as a "type wrapper"
-// for Style value type algorithms. "Type wrappers" must be simple structs with exactly one
-// member variable accessible via `get<0>(...)`.
-// NOTE: This gets automatically specialized and the get function generated when using
-// the DEFINE_STYLE_TYPE_WRAPPER macro.
-template<class> inline constexpr bool TreatAsTypeWrapper = false;
-
-#define DEFINE_STYLE_TYPE_WRAPPER(t, name) \
-    template<> inline constexpr bool TreatAsTypeWrapper<t> = true; \
-    template<size_t> const auto& get(const t& value) { return value.name; }
-
 // Types that are treated as "tuple-like" can have their conversion operations defined
 // automatically by just defining their type mapping.
-template<typename> struct ToStyleMapping;
-template<typename> struct ToCSSMapping;
+template<class> struct ToStyleMapping;
```

Comment:
Why `class` instead of `typename`?

Reply:
I don't know! I started doing it and then changed my mind. I think we should probably use `class`, as it means the same thing and is shorter (and it seems that is what the libc++ does these days), but I shouldn't be inconsistent here.

Reply:
I prefer `typename` simply because `class` is misleading. Even though it's longer.

Reply:
Fair enough. Let's stick with that then.